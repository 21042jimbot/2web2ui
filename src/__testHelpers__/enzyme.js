export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
