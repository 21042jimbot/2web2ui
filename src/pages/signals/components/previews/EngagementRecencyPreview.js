import React, { Component } from 'react';
import { Panel } from '@sparkpost/matchbox';
import { PanelLoading, PageLink } from 'src/components';
import Callout from 'src/components/callout';
import { ENGAGEMENT_RECENCY_INFO } from '../../constants/info';
import withEngagementRecencyDetails from '../../containers/EngagementRecencyDetailsContainer';
import BarChart from '../charts/barchart/BarChart';
import ChartHeader from '../ChartHeader';
import cohorts from '../../constants/cohorts';
import _ from 'lodash';
import { setSubaccountQuery } from 'src/helpers/subaccounts';
import { roundToPlaces } from 'src/helpers/units';

export class EngagementRecencyPreview extends Component {
  getYAxisProps = () => ({
    tickFormatter: (tick) => `${roundToPlaces(tick * 100, 0)}%`,
    domain: [0, 1],
    ticks: [0, 0.25, 0.5, 0.75, 1.0]
  })

  renderContent = () => {
    const { data, gap, empty, error } = this.props;

    if (error) {
      return <Callout height='170px' title='Unable to Load Data'>{error.message}</Callout>;
    }

    if (empty) {
      return <Callout height='170px' title='No Data Available'>Insufficient data to populate this chart</Callout>;
    }

    return (
      <BarChart
        height={170}
        disableHover
        margin={{ top: 12, left: 12, right: 0, bottom: 12 }}
        gap={gap}
        timeSeries={data}
        yKeys={_.keys(cohorts).map((key) => ({ key, ...cohorts[key] })).reverse()}
        yAxisProps={this.getYAxisProps()}
        xAxisProps={{ hide: true }}
      />
    );
  }

  render() {
    const { facet, facetId, loading, subaccountId } = this.props;

    if (loading) {
      return <PanelLoading minHeight='170px' />;
    }

    return (
      <PageLink to={`/signals/engagement/cohorts/${facet}/${facetId}${setSubaccountQuery(subaccountId)}`}>
        <Panel>
          <Panel.Section>
            <ChartHeader
              title='Engagement Recency'
              hideLine
              tooltipContent={ENGAGEMENT_RECENCY_INFO}
            />
          </Panel.Section>
          <Panel.Section>
            {this.renderContent()}
          </Panel.Section>
        </Panel>
      </PageLink>
    );
  }
}

export default withEngagementRecencyDetails(EngagementRecencyPreview);
