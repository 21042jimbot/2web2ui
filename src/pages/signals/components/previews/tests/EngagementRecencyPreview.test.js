import { shallow } from 'enzyme';
import React from 'react';
import { EngagementRecencyPreview } from '../EngagementRecencyPreview';

describe('Signals EngagementRecencyPreview Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      facet: 'ip-pool',
      facetId: 'best-pool',
      loading: false,
      gap: 0.5,
      empty: false,
      data: [1,2,3],
      subaccountId: 101
    };
    wrapper = shallow(<EngagementRecencyPreview {...props}/>);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders loading correctly', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty correctly', () => {
    wrapper.setProps({ empty: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders error correctly', () => {
    wrapper.setProps({ error: { message: 'error message' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('gets y axis props', () => {
    const axisProps = wrapper.find('BarChart').prop('yAxisProps');
    expect(axisProps.tickFormatter(0.2468)).toEqual('25%');
  });
});
