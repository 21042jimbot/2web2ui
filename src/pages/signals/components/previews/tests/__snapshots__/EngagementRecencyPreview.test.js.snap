// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals EngagementRecencyPreview Component renders correctly 1`] = `
<PageLink
  to="/signals/engagement-recency/ip-pool/best-pool"
>
  <Panel>
    <Panel.Section>
      <ChartHeader
        hideLine={true}
        title="Engagement Recency"
        tooltipContent="TODO"
      />
    </Panel.Section>
    <Panel.Section>
      <BarChart
        disableHover={true}
        fill="#B157CE"
        gap={0.5}
        height={170}
        margin={
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "top": 0,
          }
        }
        timeSeries={
          Array [
            1,
            2,
            3,
          ]
        }
        width="99%"
        xAxisProps={
          Object {
            "hide": true,
          }
        }
        xKey="date"
        yAxisProps={
          Object {
            "domain": Array [
              0,
              1,
            ],
            "hide": true,
          }
        }
        yKey="value"
        yKeys={
          Array [
            Object {
              "fill": "#3F8B93",
              "key": "c_new",
              "label": "New",
            },
            Object {
              "fill": "#44A8AD",
              "key": "c_14d",
              "label": "Recently Engaged",
            },
            Object {
              "fill": "#9AD8E3",
              "key": "c_90d",
              "label": "Semi Recently Engaged",
            },
            Object {
              "fill": "#B9E2E9",
              "key": "c_365d",
              "label": "Not Recently Engaged",
            },
            Object {
              "fill": "#E4EEF0",
              "key": "c_uneng",
              "label": "Never Engaged",
            },
          ]
        }
        yRange={
          Array [
            "auto",
            "auto",
          ]
        }
      />
    </Panel.Section>
  </Panel>
</PageLink>
`;

exports[`Signals EngagementRecencyPreview Component renders empty correctly 1`] = `
<PageLink
  to="/signals/engagement-recency/ip-pool/best-pool"
>
  <Panel>
    <Panel.Section>
      <ChartHeader
        hideLine={true}
        title="Engagement Recency"
        tooltipContent="TODO"
      />
    </Panel.Section>
    <Panel.Section>
      <Callout
        height="170px"
        title="No Data Available"
      >
        Insufficient data to populate this chart
      </Callout>
    </Panel.Section>
  </Panel>
</PageLink>
`;

exports[`Signals EngagementRecencyPreview Component renders error correctly 1`] = `
<PageLink
  to="/signals/engagement-recency/ip-pool/best-pool"
>
  <Panel>
    <Panel.Section>
      <ChartHeader
        hideLine={true}
        title="Engagement Recency"
        tooltipContent="TODO"
      />
    </Panel.Section>
    <Panel.Section>
      <Callout
        height="170px"
        title="Unable to Load Data"
      >
        error message
      </Callout>
    </Panel.Section>
  </Panel>
</PageLink>
`;

exports[`Signals EngagementRecencyPreview Component renders loading correctly 1`] = `
<PanelLoading
  minHeight="170px"
/>
`;
